packages/repository/src/datasource.ts

```diff
@@ -17,3 +17,11 @@ export interface DataSource {
   // tslint:disable-next-line:no-any
   [property: string]: any; // Other properties that vary by connectors
 }
+
+export interface SchemaMigrationOptions extends Options {
+  /**
+   * When set to true, schema migration will drop existing tables and recreate
+   * them from scratch, removing any existing data along the way.
+   */
+  dropExistingSchema?: boolean;
```

Comment:
Maybe instead of `boolean`, we should have an `enum`:

```ts
policyForExistingSchema: 'drop' | 'alter' | 'ignore' | 'fail';
```

Reply:
I am not sure what `ignore` would mean, but other than that I like the idea to use an enum that's easier to extend in the future.

The current juggler implementation supports `drop` and `alter` only, so that's what I am going to use in this pull request.

I find `policyForExistingSchema` too long, how about shorter `existingSchema`?

Reply:
We can start with `'drop' | 'alter'`. I'm fine with `existingSchema`.